src/utils.ts

```diff
@@ -0,0 +1,146 @@
+/*
+*  Power BI Visualizations
+*
+*  Copyright (c) Microsoft Corporation
+*  All rights reserved.
+*  MIT License
+*
+*  Permission is hereby granted, free of charge, to any person obtaining a copy
+*  of this software and associated documentation files (the ""Software""), to deal
+*  in the Software without restriction, including without limitation the rights
+*  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+*  copies of the Software, and to permit persons to whom the Software is
+*  furnished to do so, subject to the following conditions:
+*
+*  The above copyright notice and this permission notice shall be included in
+*  all copies or substantial portions of the Software.
+*
+*  THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+*  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+*  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+*  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+*  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+*  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+*  THE SOFTWARE.
+*/
+
+// d3
+import { Selection as d3Selection } from "c:/Users/grsha/OneDrive/Documents/GitHub/powerbi-visuals-sankey/node_modules/@types/d3-selection/index";
+type Selection<T> = d3Selection<any, T, any, any>;
+
+// powerbi.visuals
+import powerbi from "c:/Users/grsha/OneDrive/Documents/GitHub/powerbi-visuals-sankey/node_modules/powerbi-visuals-api/index";
+import ISelectionId = powerbi.visuals.ISelectionId;
+
+// powerbi.extensibility.utils.interactivity
+import { interactivitySelectionService, interactivityBaseService } from "c:/Users/grsha/OneDrive/Documents/GitHub/powerbi-visuals-sankey/node_modules/powerbi-visuals-utils-interactivityutils/lib/index";
+import SelectableDataPoint = interactivitySelectionService.SelectableDataPoint;
+import IInteractivityService = interactivityBaseService.IInteractivityService;
+
+import {
+SankeyDiagramNode,
+SankeyDiagramLink
+} from "c:/Users/grsha/OneDrive/Documents/GitHub/powerbi-visuals-sankey/src/dataInterfaces";
+
+const SelectedClassName: string = "selected";
+
+export function getFillOpacity(
```

Comment:
this function exists in other file in your PR.

Reply:
Utils.ts was found to be dead code so it was deleted.